src/lib/y2network/config.rb

```diff
@@ -143,7 +143,10 @@ def rename_interface(old_name, new_name, mechanism)
       interface = interfaces.by_name(old_name || new_name)
       interface.rename(new_name, mechanism)
       return unless old_name # do not modify configurations if it is just renaming mechanism
-      connections.by_interface(old_name).each { |c| c.interface = new_name }
+      connections.by_interface(old_name).each do |connection|
+        connection.interface = new_name
+        rename_dependencies(old_name, new_name, connection)
```

Comment:
np: Perhaps you could omit the `old_name` or the `new_name` as it is contained in the `connection` instance.

Reply:
where it is contained? I only found it in interface class, but not in connection one.

Reply:
Is not `connection.interface`?

Reply:
connection.interface is only the new name ( it is string ). So this way it is more explicit for me.